builtin/fsck.c

```diff
@@ -375,6 +375,7 @@ static void check_object(struct object *obj)
 static void check_connectivity(void)
```

Comment:
Looking at [object.c](https://github.com/gitgitgadget/git/blob/c16a619c5d9e2e05b6e7779311d56f3a315d9862/object.c) after this patch, it sounds as if the commit message oversells it a bit: the following functions _still_ use `the_repository`:

- `lookup_unknown_object()`
- `parse_object_or_die()`
- `clear_object_flags()` and
- `clear_commit_marks_all()`

Note: I am not saying that you should modify all of those in the same commit. Instead, I would propose to structure this PR into 6 commits, one per function in `object.c` that needs to take an `r` parameter. Each commit will then add that parameter to only one function and adjust all callers of said function, either by using an already-available `r` variable, or by introducing such a variable and initializing it to `the_repository` (these callers, unless they are in `builtin/*.c`, will have to be modified later to take an `r` parameter instead of the local variable).

Reply:
Sorry, I failed to notice the
>- clear_object_flags() 
>- clear_commit_marks_all()

functions.

Reply:
Would it be wise to make a single commit for functions `get_max_object_index` and `get_indexed_objects` given they have a complete overlap in terms of the functions calling them?

Reply:
You can do as you wish, of course, but yes, my suggestion was to structure this by function whose signature is modified.

Reply:
The changes look good now, even if I would have preferred individual commits for the two functions (it is just so much easier to review that way).

Having said that, I think the commit message needs some work. Have you looked through the commit history (e.g. via `git log -G'struct repository \*r'`) to see how previous commit messages regarding similar changes were written?

Reply:
>  Have you looked through the commit history (e.g. via `git log -G'struct repository \*r'`)

I did check previous such commits. Probably my messages need to be more comprehensive in describing their change. But since my commits are essentially doing the same stuff for all the different functions, do I have to keep repeating that message in every commit. Can you please elaborate ?

Reply:
I did not mean to imply that you should repeat a whole lot of text in all the commit messages. To the contrary, I wanted to point out that our commit messages try not to repeat what the associated diff already makes clear. In other words, they focus on the "why?" (unless obvious) over the "how?".

In that light, I would say that the following paragraph is even a bit verbose ;-)

> 'lookup_unknown_object()' and its callers are modified to enable
> passing 'r' as an argument to 'lookup_unknown_object()' in an
> effort to reduce dependence on global 'the_repository' variable.

Now, let's compare that to the (_very_ short, maybe even _too_ short) commit message of 90d3405196cfe5d3c63185a33dcee57fb6ea7e8a:

> match-trees.c: remove the_repo from shift_tree*()

My preference would be somewhere in the middle ;-) But then, my preference is also to do this one function at a time (i.e. change one function's signature per commit, adjusting all the callers in the same commit but nothing more), and as you see, the above-mentioned commit does not necessarily follow that rule.

Meaning: you have a lot of leeway how you want to structure and write your commits and commit messages.

You could even `/submit` right now :smile: